### ISSUE TITLE
Wrap MLflow integration with try block to catch exceptions.

### ISSUE THREAD

**Issue by PhilipMay** [optuna/optuna#2012]
Hi,
when using the MLflow integration it might happen that the callback wants to log something to MLflow but can not connect by http. Then an exception is thrown which cancels the optimization.

My question is: Should the MLflow calls be wrapped in an try block to catch those exceptions and just log a warning without throwing it out?

What do you think?

**Comment by PhilipMay** [comment]
Or even better: Maybe do do this change where Optuna is calling the callbacks and wrap them there.

**Comment by PhilipMay** [comment]
@drobison00 what do you think?

**Comment by PhilipMay** [comment]
I would apply the change here: https://github.com/optuna/optuna/blob/81840c2e08f452bd5ac959afaca0fee006bdb44e/optuna/_optimize.py#L170
add a try, catch exceptions and log them as a warning.

**Comment by PhilipMay** [comment]
PR made in #2017

**Comment by crcrpar** [comment]
@harupy Do you have any idea on this? I'm asking you because I think you are a committer of mlflow.

**Comment by PhilipMay** [comment]
> @harupy Do you have any idea on this? I'm asking you because I think you are a committer of mlflow.

Yes - feedback would be nice. My solution (see PR) is not targeted at MLflow but a generic solution for all exceptions in Callbacks that might happen.

**Comment by crcrpar** [comment]
Yes, your pull request would fix the error which this issue reports, but I have one comment which is now on your pull request.

**Comment by Anamitr** [comment]
I could also use such feature.
Perhaps as configuration field "treat_mlflow_errors_as_warnings".

Exp. in case of a long experiment, if MLflow tracking server goes down, I want experiment to continue, without MLflow.
